config/src/config/mod.rs

```diff
@@ -93,6 +93,7 @@ pub struct NodeConfig {
 #[serde(default, deny_unknown_fields)]
 pub struct BaseConfig {
     data_dir: PathBuf,
+    pub chain_id: ChainId,
```

Comment:
Should we just make this a string? What benefit do we have by making ChainId a struct?

Reply:
The idiom of the newtype struct is to prevent from clashing between different "strings" (https://doc.rust-lang.org/stable/rust-by-example/generics/new_types.html)

It could just be a string, since it's simply an identifier.